sycl/test-e2e/Regression/swizzle_opassign.cpp

```diff
@@ -0,0 +1,78 @@
+// REQUIRES: aspect-usm_shared_allocations
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+// RUN: %if preview-breaking-changes-supported %{  %{build} -fpreview-breaking-changes -o %t2.out   %}
+// RUN: %if preview-breaking-changes-supported %{ %{run} %t2.out  %}
+
+// Tests that the mutating operators (+=, -=, ..., ++, --) on swizzles compile
+// and correctly mutate the elements in the corresponding vector.
+
+#include <sycl/sycl.hpp>
+
+constexpr size_t NumOps = 14;
+constexpr std::string_view OpNames[NumOps] = {
+    "+=", "-=",  "*=",  "/=",        "%=",        "&=",         "|=",
+    "^=", "<<=", ">>=", "prefix ++", "prefix --", "postfix ++", "prefix ++"};
```

Comment:
```suggestion
constexpr std::string_view OpNames[] = {
    "+=", "-=",  "*=",  "/=",        "%=",        "&=",         "|=",
    "^=", "<<=", ">>=", "prefix ++", "prefix --", "postfix ++", "prefix ++"};
constexpr size_t NumOps = sizeof(OpNames) / sizeof(OpNames[0]);
```

And I vaguely recall some C++ standard added that `sizeof` "magic" under `std::`, but I don't remember the name/version or if I'm dreaming :)

Reply:
Huh! Looks like [`std::size`](https://en.cppreference.com/w/cpp/iterator/size) is a thing. That's pretty neat!